### PR TITLE
Adding ADE Environment Definitions for Bicep Public Preview

### PR DESCRIPTION
**Pull request by jacksonkays**

### CODE REVIEW

File: Environment-Definitions/TerraformTemplates/AzureDataLabs_AzureML/infra/environment.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+parameters: 
+- id: "prefix"
+  name: "Prefix"
+  type: "string"
+  required: true
+- id: "postfix"
+  name: "Postfix"
+  type: "string"
+  required: true
+- id: "location"
```
Comment (j-rewerts) [inline Environment-Definitions/TerraformTemplates/AzureDataLabs_AzureML/infra/environment.yml:14]: We should be passing [this variable](https://github.com/Azure/deployment-environments/blob/custom-runner-private-preview/Runner-Images/Terraform/scripts/deploy.sh#L27C1-L32C57) in already. Do we want to just use that?

File: Environment-Definitions/TerraformTemplates/AzureDataLabs_AzureSynapse/infra/environment.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+parameters: 
+- id: "prefix"
+  name: "Prefix"
+  type: "string"
+  required: true
+- id: "postfix"
+  name: "Postfix"
+  type: "string"
+  required: true
+- id: "location"
```
Comment (j-rewerts) [inline Environment-Definitions/TerraformTemplates/AzureDataLabs_AzureSynapse/infra/environment.yml:14]: We should be passing [this variable](https://github.com/Azure/deployment-environments/blob/custom-runner-private-preview/Runner-Images/Terraform/scripts/deploy.sh#L27C1-L32C57) in already. Do we want to just use that?

File: Environment-Definitions/TerraformTemplates/FunctionApp/environment.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: FunctionApp
+summary: Creates a function app
+description: Deploys an environment using Terraform containing a Function App
+runner: terraform
+templatePath: function_app.tf
+parameters: 
+- id: "resource_name"
```
Comment (j-rewerts) [inline Environment-Definitions/TerraformTemplates/FunctionApp/environment.yaml:7]: We should be passing [this variable](https://github.com/Azure/deployment-environments/blob/custom-runner-private-preview/Runner-Images/Terraform/scripts/deploy.sh#L27C1-L32C57) in already. Do we want to just use that? Same with location below.

File: Environment-Definitions/BicepTemplates/REAME.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Bicep Deployments
```
Comment (MaxHorstmann) [inline Environment-Definitions/BicepTemplates/REAME.md:1]: typo in filename :) 
should be README.md

Comment (jacksonkays) [reply Environment-Definitions/BicepTemplates/REAME.md]: fixed!

**Review by MaxHorstmann — APPROVED** [review]
LGTM

**Pull request merged**